src/poli/objective_repository/rosetta_energy/register.py

```diff
@@ -0,0 +1,329 @@
+        clean: bool = True,
+        relax: bool = True,
+        pack: bool = True,
+        cycle: int = 3,
+        constraint_weight: float = 5.0,
+        n_threads: int = 4,
+        batch_size: int = None,
+        parallelize: bool = False,
+        num_workers: int = None,
+        evaluation_budget: int = float("inf"),
+        force_isolation: bool = False,
+    ):
+        super().__init__(
+            batch_size=batch_size,
+            parallelize=parallelize,
+            num_workers=num_workers,
+            evaluation_budget=evaluation_budget,
+        )
+        assert wildtype_pdb_path is not None
+        self.force_isolation = force_isolation
+        self.wildtype_pdb_path = wildtype_pdb_path
+        self.score_function = score_function
+        self.seed = seed
+        self.unit = unit
+        self.conversion_factor = conversion_factor
+        self.clean = clean
+        self.relax = relax
+        self.pack = pack
+        self.cycle = cycle
+        self.constraint_weight = constraint_weight
+        self.n_threads = n_threads
+
+        try:
+
+            inner_function = get_inner_function(
+                isolated_function_name="rosetta_energy__isolated",
+                class_name="RosettaEnergyIsolatedLogic",
+                module_to_import="poli.objective_repository.rosetta_energy.isolated_function",
+                force_isolation=self.force_isolation,
+                wildtype_pdb_path=self.wildtype_pdb_path,
+                score_function=self.score_function,
+                seed=self.seed,
+                unit=self.unit,
+                conversion_factor=self.conversion_factor,
+                clean=self.clean,
+                relax=self.relax,
+                pack=self.pack,
+                cycle=self.cycle,
+                constraint_weight=self.constraint_weight,
+                n_threads=self.n_threads,
+            )
+            self.inner_function = opt_in_wrapper(inner_function)
+            self.x0 = self.inner_function.x0
+
+        except ImportError:
+            # If we weren't able to import it, we can still
+            # create it in an isolated process:
+            self.inner_function = instance_function_as_isolated_process(
+                name=self.info.get_problem_name()  # The same name in `isolated_function.py`.
+            )
+
+    def _black_box(self, x: np.ndarray, context: dict = None) -> np.ndarray:
+        """
+        Computes the stability of the mutant(s) in x.
+
+        Parameters
+        ----------
+        x : np.ndarray
+            Input array of shape [b, L] containing strings.
+        context : dict, optional
+            Additional context information (default is None).
+
+        Returns
+        -------
+        y : np.ndarray
+            The stability(DDGs/REUs/DREUs) of the mutant(s) in x.
+        """
+        return self.inner_function(x, context)
+
+    # A static method that gives you access to the information.
+    @staticmethod
```

Comment:
Black boxes now have info that's dynamic instead of static. You can now specify the `info.max_sequence_length` according to the PDBs passed by the user! Check the new implementation of rasp or foldx.